dotcom-rendering/docs/architecture/027-pictures.md

```diff
@@ -0,0 +1,133 @@
+
+The overall goal of the picture rendering to ensure that we serve the correct resolution image to the user in all cases. This can involve looking at things like the page & image layout, as well as client properties like DPR.
+
+## Background Info
+
+This background info section aims to give enough information for anyone who hasn't worked a lot with picture or source elements, or source sets. Feel free to skip it if you like!
+
+DCR uses the html `<picture>` tag to render images. This offers us a advantage over regular `<img>` tags, and that is the use of `<source>` elements to further help the browser understand what size & quality image to download, saving our users bandwidth & us money.
+
+### Fastly Image Optimiser
+
+They key enabler in this is the [fastly image optimiser](https://developer.fastly.com/reference/io/). This allows us to specify image widths when requesting an image URL. for example for a given image, say `https://i.guim.co.uk/img/media/xxxxx/yyy.jpg`, we are able to specify some important image transformation properties: `width`, `quality` & `dpr` (& others).
+
+-   `width` allows us to specify the width of the image in pixels.
+-   `quality` allows us to specify how much to compress the image, 0 being very compressed, and 100 preserving the best quality possible.
+-   `dpr` or Device Pixel Ratio allows us to scale the size of the image by this number, for example using `?width=300&dpr=2` would return a 600px wide image.
+
+### Media queries (`media="query"`)
+
+Within any given `<picture>` tag, you can have multiple `<source>` child elements. Using the `media` html attribute, which uses the same syntax as CSS media queries, you can tell the browser which `<source>` element to look for an image source in.
+
+For example `<source media="min-width: 600px">` would tell the browser to pick this source if the viewport is 600px or wider.
+
+The browser will choose and stick with the first matching source element it finds, so it's important to ensure they're in the DOM in the right order.
+
+Another important media queries we use is (`(orientation: portrait)`) to check if it's a portrait device (e.g a smartphone).
+
+### Sizes (`sizes=`)
+
+The sizes html attribute acts as the translation layer between the size of the viewport and the size of the image source you'd like to pick. A good way to think about this is that, beyond a certain width, main media (inline) images never go beyond `620px` wide, and this html attribute gives us a way to communicate that to the browser.
+
+For example `<source sizes="(min-width: 660px) 620px, 100vw">` tells the browser, "Hey, if your viewport is 660px or wider, always look for an image source which is 620px wide. If not, default to an image which is the same width as 100% of the viewport width". We can provide as many sizes & queries as we want, with the `(query) px, (query) px, ... , px)` syntax, where the last argument without a query is the default if none others match.
```

Comment:
Worth noting: it only makes sense to have multiple `sizes` if you have multiple `srcset`.